src/main/java/frc/robot/subsystems/vision/Vision.java

```diff
@@ -0,0 +1,52 @@
+package frc.robot.subsystems.vision;
+
+import java.util.Optional;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Vision extends SubsystemBase {
+
+    private final VisionIO io;
+    private final PhotonCamera camera;
+    private final PhotonPoseEstimator poseEstimator;
+    private final VisionIOInputsAutoLogged inputs = new VisionIOInputsAutoLogged();
+    private Pose2d memory = new Pose2d();
+
+    public Vision(VisionIO io) {
+        this.io = io;
+        camera = new PhotonCamera("frontCam");
+        poseEstimator = new PhotonPoseEstimator(
+                /* AprilTagFieldLayout */ null,
```

Comment:
We'll want to load the `AprilTagFieldLayout` object in the same way you're currently doing in Robot.java right now